### PR TITLE
Improve application with some clean code and leaving app warning sign

### PR DESCRIPTION
**Pull request by nagyjo**
- Leaving the app need confirm.
- Area damage set easier
- Clean code.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -76,21 +77,30 @@
         <input type="number" data-ng-model="areaDamage" onFocus="this.select();" autofocus>
 
         <div class="scrollable medium-height">
```
Comment (abrhm) [inline index.html]: Use maximum-height instead medium, to fill up the empty space.

File: index.html [hunk 2]
```diff
@@ -76,21 +77,30 @@
...
+                  <tr>
+                    <th colspan="4">{{ getAreaDamageContentLabel(key) }}</th>
+                  </tr>
+                </thead>
+
+                <tbody>
+                  <tr data-ng-repeat="character in value">
+                    <td>{{ character.displayName }}</td>
+                    <td>
+                      <label><input class="radioWithLabel" type="radio" ng-model="character.damaged" value="no">Damage Ignored?</label>
```
Comment (abrhm) [inline index.html]: Use shorter label for example: None

File: index.html [hunk 3]
```diff
@@ -76,21 +77,30 @@
...
+                    </td>
+                    <td>
+                      <label><input class="radioWithLabel" type="radio" ng-model="character.damaged" value="half">Damaged Halved?</label>
```
Comment (abrhm) [inline index.html]: Use shorter label for example: Half

File: index.html [hunk 4]
```diff
@@ -76,21 +77,30 @@
...
+                    </td>
+                    <td>
+                      <label><input class="radioWithLabel" type="radio" ng-model="character.damaged" value="full">Damage Completed?</label>
```
Comment (abrhm) [inline index.html]: Use shorter label for example: Full

File: static/app.js [hunk 5]
```diff
@@ -322,7 +319,26 @@
 
 		$scope.cancel = function() {
 			$modalInstance.close();
-		}
+		};
+
+		$scope.getAreaDamageContentLabel = function (key) {
+			if(eval(key)) {
```
Comment (abrhm) [inline static/app.js]: I think we should avoid using eval.

File: index.html [hunk 6]
```diff
@@ -76,21 +77,30 @@
...
-              <td>
-                <div class="label-wrapper" data-ng-click="toggleDamageHalved(character)" data-ng-disabled="!character.damaged">
-                  <input type="checkbox" data-ng-disabled="!character.damaged" data-ng-click="toggleDamageHalved(character)" data-ng-model="character.damageHalved"><label>Halved Damage?</label>
-                </div>
-              </td>
-            </tr>
-          </table>
+          <div data-ng-repeat="(key, value) in (cloneCharacters | groupBy: 'npc')">
+              <table class="charactersInAreaDamage">
+                <thead data-ng-click="toggleVisibility($event)">
```
Comment (abrhm) [inline index.html]: This function call can be avoid something like this:

```
<thead data-ng-init="areaDamageGroupVisibility[key] = true" data-ng-click="areaDamageGroupVisibility[key] = !areaDamageGroupVisibility[key]">

<tbody data-ng-show="areaDamageGroupVisibility[key]">
```

What do you think?

**Comment by nagyjo** [comment]
I hope i pushed the new changes..

**Pull request merged**